### ISSUE TITLE
json_scanner_get_unichar: assertion failed: (g_unichar_validate (uchar) || g_unichar_type (uchar) == G_UNICODE_SURROGATE)

### ISSUE THREAD

**Issue by Leopard1907** [ValveSoftware/steam-runtime#385]
#### Your system information

* Steam Runtime Version:  steam-runtime_0.20210126.2
* Distribution : Linux Mint 20.1 ( Ubuntu 20.04 based)
* Link to your full system information : 
 https://gist.github.com/Leopard1907/b32e35a965664fb7b35da5b233a09f5c
* Have you checked for system updates?: Yes
* Are you using the Steam Linux Runtime compatibility tool?: No

#### Please describe your issue in as much detail as possible:

I am not sure if this is Optimus specific or not but i also didn't see any other recent breakage reports about runtime and proton, i believe Soldier Runtime `0.20210317.0` ( on beta client and Soldier beta) broke my Optimus setup again, as in i need to disable to runtime again to make games that are set to 5.13 and Experimental to work.

In this state games doesn't even create a Proton Log also.

`STEAM_LINUX_RUNTIME_LOG`:

[slr-app379720-s22643941581be027.log](https://github.com/ValveSoftware/steam-runtime/files/6183595/slr-app379720-s22643941581be027.log)
[slr-app1174180-s900dedef8cc2d0a.log](https://github.com/ValveSoftware/steam-runtime/files/6183596/slr-app1174180-s900dedef8cc2d0a.log)

Output of `__NV_PRIME_RENDER_OFFLOAD=1 __GLX_VENDOR_LIBRARY_NAME=nvidia __VK_LAYER_NV_optimus=NVIDIA_only STEAM_LINUX_RUNTIME_LOG=1 steam`:

[Runtime Log.txt](https://github.com/ValveSoftware/steam-runtime/files/6183499/Runtime.Log.txt)

Steam client output **without** `STEAM_LINUX_RUNTIME_LOG=1`:

[output.txt](https://github.com/ValveSoftware/steam-runtime/files/6183511/output.txt)

System info:

https://gist.github.com/Leopard1907/6c535073b8514ee1071a0fe5a98f3c0e

Steam client sys info:

https://gist.github.com/Leopard1907/b32e35a965664fb7b35da5b233a09f5c

#### Steps for reproducing this issue:

1. An Optimus/multi gpu system
2. An intact Soldier Runtime beta ( not disabled)
3. Trying games with Proton 5.13 and Experimental, observe how they fail to start, then disable runtime and have working games again.

**Comment by kisak-valve** [comment]
Hello @Leopard1907, can you check if a Steam Linux Runtime log was generated when you used STEAM_LINUX_RUNTIME_LOG as described in https://github.com/ValveSoftware/steam-runtime/blob/master/doc/reporting-steamlinuxruntime-bugs.md#essential-information and share it?

**Comment by Leopard1907** [comment]
Sorry , missed that one. Added now.

**Comment by RyuzakiKK** [comment]
This is a locale issue.
Minimal reproducer:
```shell
$ mkdir -p ~/tmp/Ç
$ cp ~/.local/share/Steam/steamoverlayvulkanlayer.so ~/tmp/Ç/
$ cp ~/.local/share/vulkan/implicit_layer.d/steamoverlay_x86_64.json ~/.local/share/vulkan/implicit_layer.d/steamoverlay_x86_64_TMP.json
# Edit steamoverlay_x86_64_TMP.json "library_path" and set it to "/home/me/tmp/Ç/steamoverlayvulkanlayer.so"
```

**Comment by Leopard1907** [comment]
Should have guessed it. Turkish locale.

Both my locale and Optimus are things to blame usually. 🐸

Anyways, before that Soldier version i didn't encounter this issue.

I always use Soldier Beta channel.

**Comment by RyuzakiKK** [comment]
Yeah that's definitely a bug that needs to be fixed. I'm currently trying to find what caused it and how to avoid that.

**Comment by smcv** [comment]
@Leopard1907 Please could you attach whichever of your Vulkan JSON manifests is triggering this? I would guess it's an implicit layer manifest, maybe in `/usr/share/vulkan/implicit_layer.d` or `~/.local/share/vulkan/implicit_layer.d`, that has non-ASCII content.

**Comment by smcv** [comment]
> before that Soldier version i didn't encounter this issue

This surprises me: this looks like a json-glib bug, but we haven't updated json-glib for a while.

**Comment by Leopard1907** [comment]
```
asus@asus-FX553VD:~/.local/share/vulkan/implicit_layer.d$ ls
steamfossilize_i386.json    steamoverlay_x86_64.json
steamfossilize_x86_64.json  VK_LAYER_NV_GPU_Trace_release_public_2020_3_1.json
steamoverlay_i386.json      VK_LAYER_NV_nomad_release_public_2020_3_1.json
```

```
asus@asus-FX553VD:/usr/share/vulkan/implicit_layer.d$ ls
MangoHud.json  nvidia_layers.json  VkLayer_MESA_device_select.json
```

Also JSON retrieved via https://vulkan.gpuinfo.org/download.php tool


[GeForce GTX 1050.json.tar.gz](https://github.com/ValveSoftware/steam-runtime/files/6188898/GeForce.GTX.1050.json.tar.gz)

> This surprises me: this looks like a json-glib bug, but we haven't updated json-glib for a while.

A week ago everything was working fine. Since then i didn't mess with JSON's , with exception of my minor trial in `/usr/share/vulkan/icd.d/` . Which was basically moving Intel and lavapipe out of there to make sure NV remains as sole VLK device as to make sure they don't cause an interference.

I was trying to debug an issue about Doom 2016 VLK on Optimus laptops , with distorted graphics. When person that helped me with that discovered that is merely an issue of NV drivers fails at applying app profiles , i just moved those JSON's back to their usual place. In that time period i used multiple tools suchs as GFXreconstruct,Renderdoc and installed Vulkan SDK to my system via official PPA from here.

https://vulkan.lunarg.com/sdk/home#linux

Other than incident i reported above, i simply didn't mess with anything else.

Sorry for long post, i wanted to detail everything on my end as clearly as i can.

**Comment by RyuzakiKK** [comment]
> This surprises me: this looks like a json-glib bug, but we haven't updated json-glib for a while.

Apparently 0.20210217.0 doesn't have this issue, while 0.20210309.0 does.
I tested Scout because it was more practical, but the same should also apply to Soldier.

**Comment by smcv** [comment]
I think we've fixed the root cause for this crash, thanks for the report.

Workaround: remove your `/etc/vulkan/implicit_layer.d/renderdoc_capture.json`, or change its `library_path` to something that is entirely ASCII (it's the `İ` that is triggering this crash).

**Comment by Leopard1907** [comment]
Thanks for the fast response and workaround tip.

**Comment by smcv** [comment]
The next beta labelled 0.20210406.0 or later should fix this.

**Comment by Leopard1907** [comment]
```soldier	0.20210406.0	soldier	0.20210406.0	# com.valvesoftware.SteamRuntime.Platform-amd64,i386-soldier-runtime.tar.gz```

fixed the issue. Proton 6.3 with runtime functions properly with RDC layer installed now. Closing.

**Comment by smcv** [comment]
For completeness, I think the key thing in `VERSIONS.txt` is actually that you have `pressure-vessel 0.20210331.0` or later (this represents what's in the `pressure-vessel/` directory). This was first released as part of soldier 0.20210406.0 and scout 0.20210412.1.
